tasks/main.yml

```diff
@@ -1,2 +1,14 @@
 ---
+# By setting these environment parameters in a dict we circumvent the confusing
+# way that default(omit) is externally presented by Ansible.
+- set_fact:
+    os_manila_mount_credentials:
+      OS_CLOUD: "{{ os_manila_mount_os_config_name | default(omit) }}"
+      OS_PROJECT_DOMAIN_NAME: "{{ os_manila_mount_auth.project_domain_name | default(omit) }}"
+      OS_USER_DOMAIN_NAME: "{{ os_manila_mount_auth.user_domain_name | default(omit) }}"
+      OS_PROJECT_NAME: "{{ os_manila_mount_auth.project_name | default(omit) }}"
+      OS_USERNAME: "{{ os_manila_mount_auth.username | default(omit) }}"
+      OS_PASSWORD: "{{ os_manila_mount_auth.password | default(omit) }}"
+      OS_AUTH_URL: "{{ os_manila_mount_auth.auth_url | default(omit) }}"
```

Comment:
Do these omits work?

Reply:
Could just use `os_manila_mount_auth.auth_url | default`

Reply:
You're right, they don't work for environment directly, but do appear to work with this layer of indirection

Reply:
I'll try your suggestion, would be neater if it works.

Reply:
Oh, that's interesting. Whatever works.